File: C/bubblesort/llama-3.2-90b-vision-preview/llama-3.2-90b-vision-preview.c
#include <stdio.h>
#include <stdlib.h>

/**
 * Swaps two elements in an array.
 *
 * @param arr The array containing the elements to swap.
 * @param i The index of the first element to swap.
 * @param j The index of the second element to swap.
 */
void swap(int* arr, int i, int j) {
    int temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

/**
 * Performs the bubble sort algorithm on an array.
 *
 * @param arr The array to sort.
 * @param n The number of elements in the array.
 */
void bubble_sort(int* arr, int n) {
    for (int i = n - 1; i > 0; i--) {
        for (int j = 0; j < i; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr, j, j + 1);
            }
        }
    }
}

/**
 * Prints an array to the console.
 *
 * @param arr The array to print.
 * @param n The number of elements in the array.
 *
 * @note This function is commented out by default to meet the request.
 * To print the array, uncomment this function and the corresponding calls.
 */
void print_array(int* arr, int n) {
    // for (int i = 0; i < n; i++) {
    //     printf("%d ", arr[i]);
    // }
    // printf("\n");
}

int main() {
    int n = 1000; // Number of elements in the array
    int* arr = (int*)malloc(n * sizeof(int)); // Allocate memory for the array

    // Initialize the array with random values
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 1000;
    }

    // Sort the array using bubble sort
    bubble_sort(arr, n);

    // Print the sorted array
    // print_array(arr, n);

    free(arr); // Deallocate memory

    return 0;
}

File: Java/quicksort/llama-3.2-90b-vision-preview/Llama_3_2_90b_vision_preview.java
import java.util.Random;
public class Llama_3_2_90b_vision_preview {

    /**
     * Sorts an array of integers using the quick sort algorithm.
     *
     * @param arr the array to be sorted
     */
    public static void quickSort(int[] arr) {
        quickSort(arr, 0, arr.length - 1);
    }

    /**
     * Recursively sorts a sub-array using the quick sort algorithm.
     *
     * @param arr   the array to be sorted
     * @param low   the starting index of the sub-array
     * @param high  the ending index of the sub-array
     */
    private static void quickSort(int[] arr, int low, int high) {
        if (low < high) {
            int pivotIndex = partition(arr, low, high);
            quickSort(arr, low, pivotIndex - 1);
            quickSort(arr, pivotIndex + 1, high);
        }
    }

    /**
     * Partitions the array around a pivot element.
     *
     * @param arr   the array to be partitioned
     * @param low   the starting index of the sub-array
     * @param high  the ending index of the sub-array
     * @return the index of the pivot element
     */
    private static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];
        int i = low - 1;

        for (int j = low; j < high; j++) {
            if (arr[j] < pivot) {
                i++;
                swap(arr, i, j);
            }
        }

        swap(arr, i + 1, high);
        return i + 1;
    }

    /**
     * Swaps two elements in the array.
     *
     * @param arr the array
     * @param i   the index of the first element
     * @param j   the index of the second element
     */
    private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void main(String[] args) {
        Random random = new Random();
        int size = 1000;
        int[] arr = new int[size];

        for (int i = 0; i < size; i++) {
            arr[i] = random.nextInt(10000) + 1;
        }

        // Ordenando com QuickSort
        quickSort(arr);
    }

    /**
     * Prints the elements of the array.
     *
     * @param arr the array to be printed
     */
    private static void printArray(int[] arr) {
        for (int i : arr) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
}